### PR TITLE
Easy to use PWM func

### PR DESCRIPTION
**Pull request by whimbree**

### CODE REVIEW

File: cpp/driver/gpio_control.cpp [hunk 1]
```diff
@@ -163,4 +163,51 @@ void GPIOControl::Setup(MatrixIOBus *bus) {
   }
 }
 
+bool GPIOControl::SimpleServoAngle(float angle, uint16_t pin) {
+  if (!bus_) return false;
+  if (pin > 15) return false;
+
+  // Servo constants based on 9g ServoMotor Calibration
```
Comment (yoelrc88) [inline cpp/driver/gpio_control.cpp]: // Servo constants based on SG90 9g Micro Servo.

File: cpp/driver/gpio_control.cpp [hunk 2]
```diff
@@ -163,4 +163,51 @@ void GPIOControl::Setup(MatrixIOBus *bus) {
   }
 }
 
+bool GPIOControl::SimpleServoAngle(float angle, uint16_t pin) {
```
Comment (kdpatino) [inline cpp/driver/gpio_control.cpp]: In my opinion, the name of the function should be 9GServoAngle.

File: cpp/driver/gpio_control.cpp [hunk 3]
```diff
@@ -163,4 +163,51 @@ void GPIOControl::Setup(MatrixIOBus *bus) {
...
+  if (!bus_) return false;
+  if (pin > 15) return false;
+
+  // Servo constants based on SG90 9g Micro Servo
+  const int ServoRatio = 37.7;
+  const int ServoOffset = 1800;
+
+  const int FPGAClock = 150000000;
```
Comment (kdpatino) [inline cpp/driver/gpio_control.cpp]: The FPGA Frequency is here as a member of the class: https://github.com/matrix-io/matrix-creator-hal/blob/master/cpp/driver/matrixio_bus.cpp#L121 it's not necessary to set as a constant.

File: cpp/driver/gpio_control.cpp [hunk 4]
```diff
@@ -163,4 +163,51 @@ void GPIOControl::Setup(MatrixIOBus *bus) {
...
+  uint16_t channel = pin % 4;
+
+  bool b1 = SetPrescaler(bank, GPIOPrescaler);
+  bool b2 = Bank(bank).SetPeriod(period_counter);
+  bool b3 = Bank(bank).SetDuty(channel, duty_counter);
+
+  return (b1 && b2 && b3);
+}
+
+bool GPIOControl::SimpleSetPWM(float frequency, float percentage,
```
Comment (kdpatino) [inline cpp/driver/gpio_control.cpp]: Change the name of the function just SetPWM

File: cpp/driver/gpio_control.cpp [hunk 5]
```diff
@@ -163,4 +163,51 @@ void GPIOControl::Setup(MatrixIOBus *bus) {
...
+                               uint16_t pin) {
+  if (!bus_) return false;
+  if (pin > 15) return false;
+
+  const int FPGAClock = 150000000;
```
Comment (kdpatino) [inline cpp/driver/gpio_control.cpp]: Same as above

File: cpp/driver/gpio_control.cpp [hunk 6]
```diff
@@ -163,4 +163,51 @@ void GPIOControl::Setup(MatrixIOBus *bus) {
...
+  const uint16_t GPIOPrescaler = 0x5;
```
Comment (kdpatino) [inline cpp/driver/gpio_control.cpp]: Why do we use this value? , we should add a simple explanation.

File: cpp/driver/gpio_control.cpp (hunk 3 above)
Comment (kdpatino) [reply cpp/driver/gpio_control.cpp]: Get method:https://github.com/matrix-io/matrix-creator-hal/blob/master/cpp/driver/matrixio_bus.h#L48

File: cpp/driver/gpio_control.cpp (hunk 6 above)
Comment (yoelrc88) [reply cpp/driver/gpio_control.cpp]: Maybe something like this just explaining everything and giving the final value to `half_period_counter`

```C++

  // Servo constants based on SG90 9g Micro Servo
  const int ServoRatio = 37.7;
  const int ServoOffset = 1800;

  // FPGA CLock
  const int FPGAClock = 150000000;
  // We need 50Hz for servo, then 1/50Hz=0.02 sec
  // from : https://en.wikipedia.org/wiki/Servo_(radio_control)
  const float period = 0.02;
  // We choose a prescaler 32 to work with a lower frequency. 
  // FPGA needs instead the shift value as folows
  // 32 = 1 << 5 
  const uint16_t prescaler_value = 0x5; 

  /* Getting half_period_counter to generate 50Hz:

  GPIO clock after prescaler: 
  gpio_clock = 150000000 / 32 = 4687500
  
  Period counter: 
  Ratio btw clock period and servo 50Hz needed period
  period counter = 0.02 / ( 1 / 4687500 ) = 93750
  
  half_period_counter: 
  FPGA firmware need only half of the period counter
  half_period_counter = period_counter / 2 = 46875 

  */

  const uint16_t half_period_counter = 46875
  
  uint16_t duty_counter = (ServoRatio * angle) + ServoOffset;


```

**Comment by kdpatino** [comment]
Let's make the release notes :+1: .

**Pull request merged**